File: 10828.cpp
// 스택
#include <iostream>
#include <cstdio>
#include <cstring>
#define MAXSIZE 10000
using namespace std;

int main(void)
{
    int N, top = -1, out = 0;
    int stack[MAXSIZE], output[MAXSIZE];
    char command[10];

    scanf("%d", &N);

    for(int i=0; i<N; i++) {
        scanf("%s", command);
        if (strcmp(command, "push") == 0)
            scanf("%d", &stack[++top]);
        else if (strcmp(command, "size") == 0)
            output[out++] = top + 1;
        else if (strcmp(command, "empty") == 0)
            output[out++] = (top == -1);
        else if (strcmp(command, "top") == 0) {
            if (top == -1) output[out++] = -1;
            else output[out++] = stack[top];
        }
        else if (strcmp(command, "pop") == 0) {
            if (top == -1) output[out++] = -1;
            else output[out++] = stack[top--];
        }
    }

    if (out == 0)
        return 0;
    else {
        for(int i=0; i<out; i++)
            printf("%d\n", output[i]);
    }
    return 0;
}

File: 10828.py
import sys

input=sys.stdin.readline

N = int(input())
stack = []

for _ in range(N):
    command = input()[:-1]
    if command == "top":
        if stack != []:
            print(stack[-1])
        else:
            print(-1)
    elif command == "pop":
        if stack != []:
            print(stack.pop())
        else:
            print(-1)
    elif command == "size":
        print(len(stack))
    elif command == "empty":
        print(int(stack == []))
    else:
        item = int(command.split()[1])
        stack.append(item)